app/uk/gov/hmrc/agentsubscriptionfrontend/connectors/MappingConnector.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscriptionfrontend.connectors
+
+import java.net.URL
+
+import com.codahale.metrics.MetricRegistry
+import com.kenshoo.play.metrics.Metrics
+import javax.inject.{Inject, Named, Singleton}
+import play.api.libs.json.JsValue
+import uk.gov.hmrc.agent.kenshoo.monitoring.HttpAPIMonitor
+import uk.gov.hmrc.agentmtdidentifiers.model.Utr
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext.fromLoggingDetails
+
+import scala.concurrent.Future
+
+@Singleton
+class MappingConnector @Inject()(
+  @Named("agent-mapping-baseUrl") baseUrl: URL,
+  http: HttpGet with HttpPost with HttpPut with HttpDelete,
+  metrics: Metrics)
+    extends HttpAPIMonitor {
+  override val kenshooRegistry: MetricRegistry = metrics.defaultRegistry
+
+  private val mappingUrl = new URL(baseUrl, s"/agent-mapping/mappings")
+
+  def isEligibile(implicit hc: HeaderCarrier): Future[Boolean] =
```

Comment:
we can improve on this, and just have the PUT endpoint, that would give us the answer to this, and if store them. Rather than checking and do it later anyway. I am sure the endpoint does these checks anyway, so its just the matter of returning valid messages and handling them, rather than making multiple calls.
Its more of always assuming successful journey, and if you get a response no, then its simply a no, otherwise you get success

Reply:
Captured in tech debt ticket ASA-70